### PR TITLE
Cross-region load balancer with regional load balancers

### PR DESCRIPTION
**Pull request by asudbring**
# PR Checklist

Check these items before submitting a PR... 

[Contribution Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md)

[Best Practice Guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md)

- [ x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

*
*
*

### CODE REVIEW

**Comment by asudbring** [comment]
#sign-off

File: 101-load-balancer-cross-region/azuredeploy.json [hunk 1]
```diff
@@ -0,0 +1,890 @@
...
+    "parameters": {
+        "projectName": {
+            "type": "string",
+            "metadata": {
+                "description": "Specifies a project name that is used for generating resource names."
+            }
+        },
+        "location-cr": {
+            "type": "string",
+            "allowedValues": [
```
Comment (bmoore-msft) [inline 101-load-balancer-cross-region/azuredeploy.json:13]: why does this have allowedValues but the other locations do not?

File: 101-load-balancer-cross-region/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,890 @@
...
+                }
+            }
+        },
+        {
+            "type": "Microsoft.Compute/virtualMachines",
+            "apiVersion": "2020-06-01",
+            "name": "[concat('vm-r2','-',copyIndex(1))]",
+            "location": "[parameters('location-r2')]",
+            "zones": [
+                "[copyIndex(1)]"
```
Comment (bmoore-msft) [inline 101-load-balancer-cross-region/azuredeploy.json:510]: seems this template will fail if deployed to a region that doesn't support zones?

**Review by bmoore-msft — CHANGES_REQUESTED** [review]

File: 101-load-balancer-cross-region/azuredeploy.json (hunk 1 above)
Comment (asudbring) [reply 101-load-balancer-cross-region/azuredeploy.json:13]: This one is for the cross region load balancer, which is only available in select regions.  The other two are normal load balancers that are available everywhere.

File: 101-load-balancer-cross-region/azuredeploy.json (hunk 2 above)
Comment (asudbring) [reply 101-load-balancer-cross-region/azuredeploy.json:510]: I got this part of the template from the published template for deploying a standard load balancer which has zones.
https://docs.microsoft.com/en-us/azure/load-balancer/quickstart-load-balancer-standard-public-template

Comment (bmoore-msft) [reply 101-load-balancer-cross-region/azuredeploy.json:510]: @mumian - do you have a preference on how to handle these?  We can doc it or add allowedValues...

File: 101-load-balancer-cross-region/azuredeploy.json (hunk 1 above)
Comment (bmoore-msft) [reply 101-load-balancer-cross-region/azuredeploy.json:13]: Let's add some of that detail to the metadata description

Comment (asudbring) [reply 101-load-balancer-cross-region/azuredeploy.json:13]: Edited the metadata.json and updated the description to include that the cross region load balancer is available in limited regions.  I also added that the regional load balancers can be in any region.

File: 101-load-balancer-cross-region/azuredeploy.json (hunk 2 above)
Comment (mumian) [reply 101-load-balancer-cross-region/azuredeploy.json:510]: The article has this part - "The template deployment creates three availability zones. Availability zones are supported only in certain regions. Use one of the supported regions. If you aren't sure, enter centralus."

I will add the description to the quickstart template.

**Comment by asudbring** [comment]
@bmoore-msft @mumian 
Can this be merged?  It's been sitting out here for more than a week.

**Comment by asudbring** [comment]
@bmoore-msft @mumian
Can I get a status on getting this merged?

Thanks!

**Comment by mumian** [comment]
@bmoore-msft - I hope you were not waiting on me.  I have finished updating the other quickstart. Let me know if there is anything I need to do.

**Comment by bmoore-msft** [comment]
The template uses (requires) availability zones, which are not globally available, so we need to set allowedValues on the location params as we did for crlb.

**Comment by asudbring** [comment]
@bmoore-msft
Made requested changes, please merge.

Thank you!!

**Pull request merged**